src/main/java/com/stuypulse/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -0,0 +1,17 @@
+package com.stuypulse.robot.subsystems.swerve;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public abstract class SwerveModule extends SubsystemBase {
+    public abstract String getID();
+    public abstract Translation2d getOffset();
+    public abstract SwerveModuleState getState();
+    public abstract SwerveModulePosition getModulePosition();
+
+    public abstract void setState(SwerveModuleState state);
+
+    @Override
+    public abstract void periodic();
```

Comment:
whys this here, doesnt it extend subsystembase anyways

Reply:
Probably an accident, it extended the wrong class and I didn't realize but when I fixed it I forgot to remove that